experimental/packages/opentelemetry-sdk-web/src/SessionIdSpanProcessor.ts

```diff
@@ -0,0 +1,67 @@
+
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Context } from "@opentelemetry/api";
+import { ReadableSpan, Span, SpanProcessor } from "@opentelemetry/sdk-trace-base";
+
+const SESSION_ID_BYTES = 16;
+const SHARED_BUFFER = Buffer.allocUnsafe(SESSION_ID_BYTES);
+
+export class SessionIdSpanProcessor implements SpanProcessor {
+  private _sessionId;
+  private _idGenerator = getIdGenerator(SESSION_ID_BYTES);
+
+  constructor() {
+    this._sessionId = this._idGenerator();
+  }
+
+  onStart(span: Span, parentContext: Context): void {
+    span.setAttribute('session.id', this._sessionId);
+  }
+
+  onEnd(span: ReadableSpan): void {
+  }
+
+  forceFlush(): Promise<void> {
+    return Promise.resolve();
+  }
+
+  shutdown(): Promise<void> {
+    return Promise.resolve();
+  }
+}
+
+function getIdGenerator(bytes: number): () => string {
```

Comment:
Probably better to move [opentelemetry-sdk-trace-base/src/platform/<node|browser>/RandomIdGenerator.ts](https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-sdk-trace-base/src/platform/browser/RandomIdGenerator.ts) to core package utils so it can be used easily cross-packages?

Reply:
Removed for now.